agent/envoyextensions/extensioncommon/basic_envoy_extender.go

```diff
@@ -173,6 +173,8 @@ func (b BasicEnvoyExtender) patchTerminatingGatewayListener(config *RuntimeConfi
 			if ok {
 				filters = append(filters, newFilter)
 				patched = true
+			} else if err == nil { // NOTE: this is to prevent filter being added twice
```

Comment:
Would it be safer long-term to just put a `continue` inside the `if err !=nil` block for each one of these loops?

Reply:
Fixed and the PR description is updated.